siwe.go

```diff
@@ -130,14 +187,39 @@ func parseMessage(message string) (map[string]interface{}, error) {
 		}
 	}
 
+	if _, ok := result["domain"]; !ok {
+		return nil, &InvalidMessage{"`domain` must not be empty"}
+	}
+	domain := result["domain"].(string)
+	if ok, err := validateDomain(&domain); !ok {
+		return nil, err
+	}
+
+	if _, ok := result["uri"]; !ok {
+		return nil, &InvalidMessage{"`domain` must not be empty"}
+	}
+	uri := result["uri"].(string)
+	if _, err := validateURI(&uri); err != nil {
+		return nil, err
+	}
+
 	originalAddress := result["address"].(string)
 	parsedAddress := common.HexToAddress(originalAddress)
 	if originalAddress != parsedAddress.String() {
 		return nil, &InvalidMessage{"Address must be in EIP-55 format"}
 	}
 
 	if val, ok := result["resources"]; ok {
-		result["resources"] = strings.Split(val.(string), "\n- ")[1:]
+		resources := strings.Split(val.(string), "\n- ")[1:]
+		validateResources := make([]url.URL, len(resources))
+		for i, resource := range resources {
+			validateResource, err := url.Parse(resource)
+			if err != nil {
+				return nil, &InvalidMessage{fmt.Sprintf("Invalid format for field `resources` at position %d", i)}
+			}
+			validateResources[i] = *validateResource
+		}
+		result["resources"] = validateResources
 	}
 
 	return result, nil
```

Comment:
can we provide inline doc for public / exported functions -> ParseMessage(...): Message?